### ISSUE TITLE
Missing mosaic id object size in mosaic (Nem schema)

### ISSUE THREAD

**Issue by fboucquez** [symbol/catbuffer-schemas#21]
I've created a Mosaic object in `nem.core` and serialize it. Script:

Java
```java=
final Mosaic mosaic = Utils.createMosaic(10, 23);
final byte[] bytes = BinarySerializer.serializeToBytes(mosaic);
System.out.println("");
System.out.println(HexEncoder.getString(bytes));
System.out.println("");
System.out.println(JsonSerializer.serializeToJson(mosaic));
```

Json
```json=
{ 'quantity': 23, 'mosaicId': { 'namespaceId': 'id10', 'name': 'name10' } }
```

This the printed payload (spaced):

`12000000 04000000 69643130 06000000 6e616d653130 1700000000000000`

| Name | Size | Hex Value | Parsed Value | Remark  |
| - | - | - | - | - |
| mosaicIdSize | 4 | 12000000 | 18 | Missing in new NEM schemas! |
| namespaceIdLabelSize | 4 | 04000000 | 4 | OK |
| namespaceIdLabel | namespaceIdLabelSize(4) | 69643130 | id10 | OK |
| mosaicIdLabelSize | 4 | 06000000 | 6 | OK |
| mosaicIdLabel | mosaicIdLabelSize(6) | 6e616d653130 | name10 | OK |
| quantity | 8 | 1700000000000000 | 23 | OK |

Note, the current schema works with a payload like `04000000 69643130 06000000 6e616d653130 1700000000000000`, when the mosiac id size is removed

The size of the object mosaic_id is missing around here:
https://github.com/symbol/catbuffer-schemas/blob/main/nem/mosaic.cats#L22

This object size may or may not need a catparser upgrade as far as I can tell. It cannot be reserved as the value is dynamic, the length of the inline object (mosaicId).

**Comment by Jaguar0625** [comment]
good find. i've confirmed that this is indeed a bug. 

the best we could do now, without extending the schema, is something like this:
```
    # mosaic id size
    mosaic_id_size = make_reserved(uint32, 8)  # add size and namespace_id.size to constant
```

we could potentially extend the schema to support something like this:
```
    # mosaic id size
    mosaic_id_size = make_reserved(uint32, [8, size, namespace_id.size])
```

**Comment by Jaguar0625** [comment]
fixed by #49
